xc/device.mli

```diff
@@ -80,7 +80,8 @@ sig
 	       -> mac:string -> carrier:bool 
 	       -> ?mtu:int -> ?rate:(int64 * int64) option
 	       -> ?protocol:protocol -> ?backend_domid:Xenctrl.domid 
-	       -> ?other_config:((string * string) list) 
+	       -> ?other_config:((string * string) list)
+	       -> ?static_ip_setting:((string * string) list)
```

Comment:
Hi @robhoes ,

It looks like extra_private_keys is placed under /xapi/<dom uuid>/private xenstore path. DomU has no permission to RW this xenstore path.

However, the static ip setting must be read by DomU guest agent process.

And actually the xenstore path where to save static ip setting is not easy to decide, please see http://xenbits.xen.org/gitweb/?p=xen.git;a=blob;f=docs/misc/xenstore-paths.markdown, there are some specifications on the xenstore path.